### PR TITLE
`IfStatement` and `IfExpression`: clean up acceptable block syntax irregularities

### PR DESCRIPTION
**Pull request by wcjohnson**
- `parseIfStatement`:
  - Add parameter to enforce matching whiteblock syntax
  - Check for matching whiteblock syntax

- Added test fixtures for `IfStatement`s

- Ignore npm-debug.log files

### CODE REVIEW

**Comment by wcjohnson** [comment]
I believe this is fully correct as concerns `IfStatement`s, and is therefore mergeable on its own. All tests are passing.

It doesn't cover `IfExpression`s but as discussed, those need a little more work.

**Comment by wcjohnson** [comment]
@rattrayalex 

Updated this according to your suggestion. Might be the best effort so far, but still one outstanding issue. You said you wanted this to compile:
```
f = ->
  if (true) return 1 else return 2
```
It doesn't:
```
Unexpected token, expected ; (2:21)
```
FWIW, this also fails to compile in vanilla JS and currently-live LS, for the same reason.

I pulled on the thread a little bit, disabling the semicolon requirement in `parseReturnStatement`, and the result was 10 mainline test failures. None of them looked particularly serious -- a few of them were just expectations of failure -- but some of them had to do with flow typing and I'm not sure it's a can of worms I want to open.

This also fails to compile for the same reason:
```
f = ->
  if true: return 1 else: return 2
```

I've updated the PR with new test fixtures which resign themselves to the failure, but if you want that to compile you are probably going to have to open the can of worms yourself.

Other than that, though, everything seems to be working well re `IfStatements`.

File: test/fixtures/lightscript/if-statement/oneline-return-colons/actual.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+f = ->
```
Comment (rattrayalex) [inline test/fixtures/lightscript/if-statement/oneline-return-colons/actual.js]: let's use idiomatic lsc in the tests: `f() ->`

File: test/fixtures/lightscript/if-statement/oneline-return-colons/actual.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+  if true: return 1 else: return 2
```
Comment (rattrayalex) [inline test/fixtures/lightscript/if-statement/oneline-return-colons/actual.js]: I don't think we need a test for this... was a silly mistake on my part to suggest. 

```
if true: 1 else: 2
```
and/or
```
if true: return 1
else: return 2
```
might make more sense instead. But not sure either of those are necessary

File: test/fixtures/lightscript/if-statement/oneline-return-undelimited/actual.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+f = ->
+  if (true) return 1 else return 2
```
Comment (rattrayalex) [inline test/fixtures/lightscript/if-statement/oneline-return-undelimited/actual.js]: yeah, sorry, again this was my bad – let's test the other forms here:
```
if (true) return 1 
else return 2
```
and
```
if (true) 1 else 2
```
both of which should be legal, if discouraged. 

fine to leave this test too

**Review by rattrayalex — COMMENTED** [review]
I think this looks good – couple tests that could be added, and I think there are some redundant tests as well, but happy to accept this as-is

File: test/fixtures/lightscript/if-statement/oneline-return-undelimited/actual.js (hunk 3 above)
Comment (rattrayalex) [reply test/fixtures/lightscript/if-statement/oneline-return-undelimited/actual.js]: Oops I'm an idiot the latter isn't possible; I mean
```
if (true) 1; else 2
```
but that's redundant with the first example so not worth testing

**Comment by wcjohnson** [comment]
Updated as requested.

**Comment by rattrayalex** [comment]
Awesome, thanks! I think this looks good. Shall we merge in and I can separately add my branch with If-Expressions? I'll probably want to squash those commits together, FYI

**Comment by rattrayalex** [comment]
(you'd be more than welcome to add the patch from my branch if you like)

**Comment by wcjohnson** [comment]
I have another branch (Based on this one) that adds your patch as well. Let me merge it.

**Comment by wcjohnson** [comment]
OK, both changes are on this branch now, and there are some matching test fixtures under the `if-expressions` as well.

**Comment by rattrayalex** [comment]
Awesome work!

**Pull request merged**